cdk/src/open-data-platform/app-plane/api/api-monitoring.ts

```diff
@@ -0,0 +1,78 @@
+import { Construct } from 'constructs';
+import { RestApi } from 'aws-cdk-lib/aws-apigateway';
+import { Duration } from 'aws-cdk-lib';
+import { ITopic } from 'aws-cdk-lib/aws-sns';
+import * as cloudwatch from 'aws-cdk-lib/aws-cloudwatch';
+import { SnsAction } from 'aws-cdk-lib/aws-cloudwatch-actions';
+
+interface ApiMonitoringProps {
+  gateway: RestApi;
+  ticketSNSTopic?: ITopic;
+}
+
+export class ApiMonitoring extends Construct {
+  constructor(scope: Construct, id: string, props: ApiMonitoringProps) {
+    super(scope, id);
+
+    const { gateway, ticketSNSTopic } = props;
+
+    // Client errors are likely caused by problems in our client code since the API is not public,
+    // so they should be considered like an internal error.
+    // TODO: Split this into client and server errors once the API is pubished.
+    // TODO: Track the error budget, rather than looking at the error rate at any given time.
+
+    const availabilitySlo = 0.95;
+    const latencyMsSlo = 2000; // Miliseconds
+    const slowBurnPeriod = Duration.days(1);
+    const fastBurnPeriod = Duration.hours(1);
+
+    const alarms = [
+      new cloudwatch.MathExpression({
+        expression: '(clientError + serverError) / count',
+        label: 'Error Fraction',
+        period: slowBurnPeriod,
+        usingMetrics: {
+          clientError: gateway.metricClientError(),
+          serverError: gateway.metricServerError(),
```

Comment:
would it be worth having a separate alert for client vs server errors? I agree that alerting here for client errors makes sense for now, but it seems like these would probably be different issues so having both add up to 5% of total metrics might not be as significant of having 5% be just server or client errors (seems more likely the latter would signify a larger issue)

Reply:
Yeah I think that's a good idea. As-is, this will be more sensitive than separating it out. I want to see how much this alarm goes off then we can tune (and possibly separate it). I have a TODO for that.